internal/tofu/context_plugins.go

```diff
@@ -148,9 +144,12 @@ func (cp *contextPlugins) ResourceTypeSchema(ctx context.Context, providerAddr a
 // ProvisionerSchema uses a temporary instance of the provisioner with the
 // given type name to obtain the schema for that provisioner's configuration.
 //
-// ProvisionerSchema memoizes results by provisioner type name, so it's fine
-// to repeatedly call this method with the same name if various different
-// parts of OpenTofu all need the same schema information.
+// Provisioner schemas are currently not cached because we assume that it's
+// rare to use any except those compiled directly into OpenTofu, and therefore
+// we're usually just retrieving an already-resident data structure from a
+// different part of the program. This could potentially be slow for those
+// using the legacy support for plugin-based provisioners, if they have many
+// instances of such provisioners.
```

Comment:
This is just a correction of the comment and not a change in behavior: this function was already not doing any memoization itself, despite what this comment claimed.